coordinator/src/main/scala/filodb.coordinator/ShardManager.scala

```diff
@@ -366,7 +367,7 @@ private[coordinator] final class ShardManager(settings: FilodbSettings,
         val metrics = new ShardHealthStats(setup.ref, _shardMappers(dataset.ref))
         val resources = setup.resources
         val source = setup.source
-        val state = DatasetInfo(resources, metrics, source, setup.storeConfig, dataset)
+        val state = DatasetInfo(resources, metrics, source, setup.downsample, setup.storeConfig, dataset)
```

Comment:
Yes. We need to do this sooner or later with all the new features (like histograms) flowing in.
It may just be easier to get this all in one document and place it either in configuration file. There is much scope for and need to simplify the dataset configuration loading part quite a bit.

Deferred to another PR.

Reply:
Ok can we add a TODO and issue

Reply:
What I was trying to get at was that there were so many changes to so many files due to adding downsampling config, that it would have been easier to just change it into one config.